bundle/src/main/java/com/adobe/acs/commons/redirects/filter/RedirectFilter.java

```diff
@@ -0,0 +1,508 @@
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventConstants;
+import org.osgi.service.event.EventHandler;
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.AttributeType;
+import org.osgi.service.metatype.annotations.Designate;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.management.NotCompliantMBeanException;
+import javax.management.openmbean.*;
+import javax.servlet.*;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static org.apache.sling.engine.EngineConstants.SLING_FILTER_SCOPE;
+import static org.osgi.framework.Constants.SERVICE_DESCRIPTION;
+import static org.osgi.framework.Constants.SERVICE_RANKING;
+
+/**
+ * A request filter that implements support for virtual redirects.
+ */
+@Component(service = {Filter.class, RedirectFilterMBean.class, EventHandler.class}, property = {
+        SERVICE_DESCRIPTION + "=A request filter implementing support for virtual redirects",
+        SLING_FILTER_SCOPE + "=" + EngineConstants.FILTER_SCOPE_REQUEST,
+        SERVICE_RANKING + ":Integer=10000",
+        "jmx.objectname=" + "com.adobe.acs.commons:type=Redirect Manager",
+        EventConstants.EVENT_TOPIC + "=" + ReplicationAction.EVENT_TOPIC,
+        ResourceChangeListener.CHANGES + "={ADDED, REMOVED, CHANGED}"
+
+})
+@Designate(ocd = RedirectFilter.Configuration.class)
+public class RedirectFilter extends AnnotatedStandardMBean
+        implements Filter, EventHandler, ResourceChangeListener, RedirectFilterMBean {
+
+    public static final String DEFAULT_STORAGE_PATH = "/var/acs-commons/redirects";
+    public static final String ACS_REDIRECTS_RESOURCE_TYPE = "acs-commons/components/utilities/manage-redirects";
+    public static final String REDIRECT_RULE_RESOURCE_TYPE = ACS_REDIRECTS_RESOURCE_TYPE + "/redirect-row";
+
+
+    private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    static final String SERVICE_NAME = "redirect-manager";
+
+    @ObjectClassDefinition(name = "ACS Commons Redirect Filter")
+    public @interface Configuration {
+        @AttributeDefinition(name = "Enable Redirect Filter", description = "Indicates whether the redirect filter is enabled or not.", type = AttributeType.BOOLEAN)
+        boolean enabled() default true;
+
+        @AttributeDefinition(name = "Rewrite Location Header", description = "Apply Sling Resource Mappings (/etc/map) to Location header. " +
+                "Use if Location header should rewritten using ResourceResolver#map", type = AttributeType.BOOLEAN)
+        boolean rewriteUrls() default true;

```

Comment:
mapUrls instead of rewriteUrls as rewrite is a bit too generic

Reply:
renamed to mapUrls()